client/src/hooks/useEventMember.ts

```diff
@@ -23,129 +23,102 @@ const useEventMember = (): ReturnUseEventMember => {
 
   const [reports, setReports] = useState<Report[]>(initialReports);
   const [deleteMembers, setDeleteMembers] = useState<number[]>([]);
-  const [changedMembers, setChangedMembers] = useState<MemberWithDeposited[]>([]);
-  const [isCanRequest, setIsCanRequest] = useState<boolean>(false);
-  const [filteredChangedMembers, setFilteredChangedMembers] = useState<MemberWithDeposited[]>([]);
 
   useEffect(() => {
     setReports(initialReports);
   }, [initialReports]);
 
-  useEffect(() => {
-    const changedMembers = getChangedMembers();
+  const isCanSubmit = useMemo(() => {
+    // 중복되는 이름이 존재하는지 확인
+    const hasDuplicateMemberName = (): boolean => {
+      const nameSet = new Set(reports.map(member => member.memberName));
+      return nameSet.size !== reports.length;
+    };
 
-    // 중복된 이름이 존재할 경우 isCanRequest를 false로 변경
     if (hasDuplicateMemberName()) {
-      setIsCanRequest(false);
-    } else {
-      // 변경된 사항이 존재하거나 삭제한 member가 존재한다면 isCanRequest를 true로 변경
-      setIsCanRequest(changedMembers.length > 0 || deleteMembers.length > 0);
+      // 중복 이름이라면 false
+      return false;
     }
 
-    // memberName 유효성 검사 (0글자)
-    const hasEmptyName = changedMembers.some(member => member.name.trim().length === 0);
-    if (hasEmptyName) setIsCanRequest(false);
+    // 이름이 공백이라면 false
+    const hasEmptyName = reports.some(report => report.memberName.trim().length === 0);
+    if (hasEmptyName) return false;
+
+    // 초기 값과 비교하여 변경된 사항이 존재하는지 확인
+    const hasChanges = reports.some(report =>
+      initialReports.find(
+        initial =>
+          initial.memberId === report.memberId &&
+          (initial.memberName !== report.memberName || initial.isDeposited !== report.isDeposited),
+      ),
+    );
 
-    setFilteredChangedMembers(changedMembers);
-  }, [reports, changedMembers, deleteMembers]);
+    // 변경된 사항이 존재 혹은 삭제된 member가 존재한다면 true
+    return hasChanges || deleteMembers.length > 0;
+  }, [reports, initialReports, deleteMembers]);
 
-  const changeMemberName = (memberId: number, newName: string) => {
-    // 유효성 검사 (4글자)
+  const changeMemberName = useCallback((memberId: number, newName: string) => {
+    // 유효성 검사 (4자 이하)
     if (!validateMemberName(newName).isValid) {
       return;
     }
 
     setReports(prevReports =>
       prevReports.map(report => (report.memberId === memberId ? {...report, memberName: newName} : report)),
     );
+  }, []);
```

Comment:
두 함수가 클라이언트 실행 중 변경될 일은 없겠다만, 안정성을 위해서
setReport, validateMenberName이 변경될 때는 캐싱된 콜백함수가 변경되어야 하지 않을까요?

```suggestion
  }, [setReport, validateMemberName]);
```